File: Hash/Medium/Lint793-Intersection-of-Arrays/solution_hashTable.cpp
class Solution {
public:
    /**
     * @param arrs: the arrays
     * @return: the number of the intersection of the arrays
     */
    int intersectionOfArrays(vector<vector<int>> &arrs) {
        // hash table
        // tc:O(m*n); sc:O(m*n)
        if(arrs.size() == 0) {
            return 0;
        }
        unordered_map<int, int> dict;
        for(auto arr : arrs) {
            for(auto num : arr) {
                dict[num] = dict.find(num) != dict.end() ? ++dict[num] : 1;
            }
        }
        int res = 0;
        for(auto it = dict.begin(); it != dict.end(); it++) {
            if(it->second == arrs.size()) {
                res++;
            } 
        }
        return res;
    }
};

File: Hash/Easy/1-Two-Sum/solution_hashtable.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // hash table
        // tc:O(n); sc:O(n)
        if(nums == null || nums.Length < 2) {
            return new int[0];
        }
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++) {
            int num = target - nums[i];
            if(dict.ContainsKey(num)) {
                return new int[] { dict[num], i };
            }
            dict[nums[i]] = i;
        }
        return new int[0];
    }
}